src/backend/settings.rs

```diff
@@ -0,0 +1,101 @@
+use async_trait::async_trait;
+use futures_channel::{
+    mpsc::{Receiver, Sender},
+    oneshot,
+};
+use futures_util::{FutureExt, StreamExt};
+use std::{cell::RefCell, collections::HashMap};
+use zbus::dbus_interface;
+
+use crate::{
+    backend::{Backend, IMPL_PATH},
+    desktop::settings::Namespace,
+    zvariant::OwnedValue,
+};
+
+#[async_trait]
+pub trait SettingsImpl {
+    async fn read_all(&self, namespaces: Vec<String>) -> HashMap<String, Namespace>;
+
+    async fn read(&self, namespace: &str, key: &str) -> OwnedValue;
+}
+
+pub struct Settings<T: SettingsImpl> {
+    receiver: RefCell<Option<Receiver<Action>>>,
+    imp: T,
+}
+
+unsafe impl<T: Send + SettingsImpl> Send for Settings<T> {}
+unsafe impl<T: Sync + SettingsImpl> Sync for Settings<T> {}
+
+impl<T: SettingsImpl> Settings<T> {
+    pub async fn new(imp: T, backend: &Backend) -> zbus::Result<Self> {
+        let (sender, receiver) = futures_channel::mpsc::channel(10);
+        let iface = SettingsInterface::new(sender);
+        let object_server = backend.cnx().object_server();
+
+        object_server.at(IMPL_PATH, iface).await?;
```

Comment:
```suggestion
        let object_server = backend.cnx().object_server();

        object_server.at(IMPL_PATH, iface).await?;
       
```
this seems to be repeated everywhere, can you add a helper method for it to the Backend type that would do the necessary calls for you?